finagle-thrift/src/main/ruby/lib/finagle-thrift/trace.rb

```diff
@@ -100,13 +100,14 @@ class Flags
   end
 
   class SpanId
-    HEX_REGEX = /^[a-f0-9]{16}$/i
+    HEX_REGEX = /^[a-f0-9]{16,32}$/i
```

Comment:
I think we are good and this will work properly for us. It will work as expected, will read a string (which comes from the header) and create a SpanId class from it.